x-pack/test/apm_api_integration/tests/mobile/mobile_most_used_chart.spec.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { APIReturnType } from '@kbn/apm-plugin/public/services/rest/create_call_apm_api';
+import { ENVIRONMENT_ALL } from '@kbn/apm-plugin/common/environment_filter_values';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { generateMobileData } from './generate_mobile_data';
+
+type MostUsedCharts =
+  APIReturnType<'GET /internal/apm/mobile-services/{serviceName}/most_used_charts'>;
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const apmApiClient = getService('apmApiClient');
+  const registry = getService('registry');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  const start = new Date('2023-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2023-01-01T00:15:00.000Z').getTime() - 1;
+
+  async function getMobileMostUsedCharts({
+    environment = ENVIRONMENT_ALL.value,
+    kuery = '',
+    serviceName,
+    transactionType = 'mobile',
+  }: {
+    environment?: string;
+    kuery?: string;
+    serviceName: string;
+    transactionType?: string;
+  }) {
+    return await apmApiClient
+      .readUser({
+        endpoint: 'GET /internal/apm/mobile-services/{serviceName}/most_used_charts',
+        params: {
+          path: { serviceName },
+          query: {
+            environment,
+            start: new Date(start).toISOString(),
+            end: new Date(end).toISOString(),
+            kuery,
+            transactionType,
+          },
+        },
+      })
+      .then(({ body }) => body);
+  }
+
+  registry.when(
+    'Most used charts when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      describe('when no data', () => {
+        it('handles empty state', async () => {
+          const response: MostUsedCharts = await getMobileMostUsedCharts({ serviceName: 'foo' });
+          expect(response.mostUsedCharts.length).to.eql(4);
```

Comment:
[nit] is this assertion relevant?

Reply:
I wanted to make this test a bit more specific by saying that the API must return all 4 chart data as behind the scenes it makes 2 API calls to retrieve 1 and 3 charts respectively.

Reply:
Couldn't we just test that the two calls have been made? (don't know if that's possible for these type of tests)

Reply:
I feel that would fall more on the Unit Test side rather than the High level API test